rules/DowngradePhp80/Rector/Class_/DowngradePropertyPromotionRector.php

```diff
@@ -79,17 +79,18 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Node
     {
-        $oldComments = $this->getOldComments($node);
-        $promotedParams = $this->resolvePromotedParams($node);
-        if ($promotedParams === []) {
+        $constructorClassMethod = $node->getMethod(MethodName::CONSTRUCT);
+        if (! $constructorClassMethod instanceof ClassMethod) {
             return null;
         }
 
-        /** @var ClassMethod $constructClassMethod */
-        $constructClassMethod = $node->getMethod(MethodName::CONSTRUCT);
-
-        $properties = $this->resolvePropertiesFromPromotedParams($constructClassMethod, $promotedParams, $node);
+        $oldComments = $this->getOldComments($constructorClassMethod);
```

Comment:
`oldComments` definition can be moved below if return early

Reply:
comments will be modified in `resolvePromotedParams()` therefore the "old" comments need to be stored before